fn/slice.go

```diff
@@ -205,3 +226,24 @@ func Sum[B Number](items []B) B {
 func HasDuplicates[A comparable](items []A) bool {
 	return len(NewSet(items...)) != len(items)
 }
+
+// ForEachConc maps the argument function over the slice, spawning a new
+// goroutine for each element in the slice and then awaits all results before
+// returning them.
+func ForEachConc[A, B any](f func(A) B, as []A) []B {
+	wait := sync.WaitGroup{}
+	bs := make([]B, len(as))
+
+	for i, a := range as {
+		i, a := i, a
```

Comment:
not sure I understand this swap

Reply:
What swap? I think this is just the required loopvar aliasing because go is strange in this way.

Reply:
We can drop this safely now (latest Go version).